### PR TITLE
Fix handling of arrays in two-argument `ranges::distance`

### PR DESCRIPTION
**Pull request by CaseyCarter**
.... which was broken by LWG-3392. This implements the proposed resolution of a submitted-but-not-yet-numbered LWG issue.

### CODE REVIEW

File: stl/inc/xutility [hunk 1]
```diff
@@ -2843,16 +2843,16 @@ namespace ranges {
         using _Not_quite_object::_Not_quite_object;
 
         // clang-format off
-        template <input_or_output_iterator _It, sentinel_for<_It> _Se>
+        template <class _It, sentinel_for<_It> _Se>
```
Comment (miscco) [inline stl/inc/xutility:2846]: You really do not like the iterator concept

**Review by miscco — APPROVED** [review]

File: tests/std/tests/P0896R4_ranges_iterator_machinery/test.cpp [hunk 2]
```diff
@@ -3474,9 +3474,19 @@ namespace vso1121031 {
     STATIC_ASSERT(!has_member_value_type<indirectly_readable_traits<iterish<float const volatile>>>);
 } // namespace vso1121031
 
+constexpr bool test_lwg_XXX() {
+    int a[3] = {};
```
Comment (miscco) [inline tests/std/tests/P0896R4_ranges_iterator_machinery/test.cpp:3478]: Maybe add another const array for my paranoia

File: stl/inc/xutility (hunk 1 above)
Comment (CaseyCarter) [reply stl/inc/xutility:2846]: Ironically enough removing the type-constraints on the iterator parameters came from @Quuxplusone, unprompted by me.

File: tests/std/tests/P0896R4_ranges_iterator_machinery/test.cpp [hunk 3]
```diff
@@ -3474,9 +3474,19 @@ namespace vso1121031 {
     STATIC_ASSERT(!has_member_value_type<indirectly_readable_traits<iterish<float const volatile>>>);
 } // namespace vso1121031
 
+constexpr bool test_lwg_XXX() {
```
Comment (CaseyCarter) [inline tests/std/tests/P0896R4_ranges_iterator_machinery/test.cpp:3477]: I'd like to annotate (here and in the header) with the proper issue number before merging.

**Comment by CaseyCarter** [comment]
Converted to draft after Tim pointed out some issues with the PR.

**Comment by CaseyCarter** [comment]
This PR was superseded by #2522.

**Pull request closed without merging**